rdkit/Chem/Fraggle/FraggleSim.py

```diff
@@ -29,356 +29,285 @@
+          result = f
+    return result
+
+  elif (ftype == FTYPE_CYCLIC_ACYCLIC):
+    result = []
+
+    # need to find the fragments which are valid which means they must be:
     #  Terminal (one attachment point) or valid ring cut
     for f in fragments:
       attachments = f.count("*")
 
-      if (attachments >= 3):
+      if attachments >= 3:
         continue
 
-      if (attachments == 2):
-        #check if a valid cut
-        valid, result_hcount = is_ring_cut_valid(f)
-        if (valid):
-          #needs to be greater 3 heavy atoms
-          if (result_hcount > 3):
-            result = "%s.%s" % (result, f)
+      if attachments == 2:
+        # check if a valid cut
+        valid, fhac = is_ring_cut_valid(f)
+        # needs to be greater 3 heavy atoms
+        if valid and fhac > 3:
+          result.append(f)
 
-      elif (attachments == 1):
+      elif attachments == 1:
         fMol = Chem.MolFromSmiles(f)
         fhac = fMol.GetNumAtoms()
-        #needs to be greater 3 heavy atoms
-        if (fhac > 3):
-          result = "%s.%s" % (result, f)
-          result_hcount = result_hcount + fhac
-
-    #print "F: %s" % (result)
-
-    #appropriate fragmentations must have 2 components
-    #result will always start with . because of the way it is constructed
-    #hence 2 component result wil contain 2 dots
-    if ((result != "") and (result.count(".") == 2)):
-      #take off the starting dot when building smiles
-      fMol = Chem.MolFromSmiles(result[1:])
+        # needs to be greater 3 heavy atoms
+        if fhac > 3:
+          result.append(f)
+
+    # appropriate fragmentations must have 2 components with more than 3 heavy atoms.
+    if len(result) == 2:
+      result = '.'.join(result)
+      fMol = Chem.MolFromSmiles(result)
       result_hcount = fMol.GetNumAtoms()
-      #needs to be greater 3 heavy atoms and greater than 60% of parent mol
-      if ((result_hcount > 3) and (result_hcount > 0.6 * hac)):
-        #take off the starting dot
-        result = result[1:]
-      else:
+      # needs to be greater than 60% of parent mol
+      if fMol.GetNumAtoms() <= 0.6 * hac:
         result = None
     else:
       result = None
+    return result
 
-  return result
-
-#Global smarts used by the program
-#acyclic bond smarts
-acyc_smarts = Chem.MolFromSmarts("[*]!@!=!#[*]")
-
-#exocyclic/fused exocyclic bond smarts
-cyc_smarts = Chem.MolFromSmarts("[R1,R2]@[r;!R1]")
-
-#smarts used to find appropriate fragment for
-#would use SMARTS: [$([#0][r].[r][#0]),$([#0][r][#0])]
-#but rdkit doesn't support component SMARTS in recursive one - $([#0][r].[r][#0])
-#hence split into two
-cSma1 = Chem.MolFromSmarts("[#0][r].[r][#0]")
-cSma2 = Chem.MolFromSmarts("[#0][r][#0]")
+  else:
+    raise NotImplementedError('Invalid fragmentation type {0}'.format(ftype))
 
 
-def generate_fraggle_fragmentation(mol):
-  """
-    >>> q = Chem.MolFromSmiles('COc1cc(CN2CCC(NC(=O)c3cncc(C)c3)CC2)c(OC)c2ccccc12')
```

Comment:
I would prefer to leave this documentation in here because it makes it very clear what the return looks like.

We can argue about whether or not we run the doctests, but that's a separate point